diff -Naur newlib-2.4.0-standard/config.sub newlib-2.4.0/config.sub
--- newlib-2.4.0-standard/config.sub	2016-03-29 21:33:42 +0000
+++ newlib-2.4.0/config.sub	2016-08-11 21:55:23 +0000
@@ -1377,6 +1377,7 @@
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
+	      | -*Sharpen* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* | -cloudabi* | -sortix* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff -Naur newlib-2.4.0-standard/newlib/configure.host newlib-2.4.0/newlib/configure.host
--- newlib-2.4.0-standard/newlib/configure.host	2016-03-29 21:33:42 +0000
+++ newlib-2.4.0/newlib/configure.host	2016-08-11 21:57:35 +0000
@@ -480,7 +480,10 @@
 	##newlib_cflags="${newlib_cflags} -I`newlib-flags --kernel-dir`/include"
 	##newlib_cflags="${newlib_cflags} -idirafter ${gcc_dir}include"
 	;;
-
+  i[3-7]86-*-Sharpen*)
+	sys_dir=Sharpen
+	newlib_cflags="${newlib_cflags} -DSIGNAL_PROVIDED"
+	;;
   m68hc11-*-*|m6811-*-*|m6812-*-*|m68hc12-*-*)
 	;;
diff -Naur newlib-2.4.0-standard/newlib/libc/sys/configure.in newlib-2.4.0/newlib/libc/sys/configure.in
--- newlib-2.4.0-standard/newlib/libc/sys/configure.in	2016-03-29 21:33:42 +0000
+++ newlib-2.4.0/newlib/libc/sys/configure.in	2016-08-11 21:58:19 +0000
@@ -30,6 +30,7 @@
 	h8300hms) AC_CONFIG_SUBDIRS(h8300hms) ;;
 	h8500hms) AC_CONFIG_SUBDIRS(h8500hms) ;;
 	linux) AC_CONFIG_SUBDIRS(linux) ;;
+	Sharpen) AC_CONFIG_SUBDIRS(Sharpen) ;;
 	m88kbug) AC_CONFIG_SUBDIRS(m88kbug) ;;
 	mmixware) AC_CONFIG_SUBDIRS(mmixware) ;;
 	netware) AC_CONFIG_SUBDIRS(netware) ;;
